### PR TITLE
show color map translation complete.

### PR DESCRIPTION
**Pull request by brigb123**
this adds XY translation options to ShowColorMapModule.cc . This replaces "Add Extra Space" and allows the user to translate the scale bar from 0-100% of the screen width/height. 

When the bar passes 50%, The text switches sides of the bar for visible convenience.

### CODE REVIEW

File: src/Modules/Visualization/ShowColorMapModule.cc [hunk 1]
```diff
@@ -210,14 +210,16 @@ ShowColorMapModule::buildGeometryObject(ColorMapHandle cm, ModuleStateHandle sta
       cm->getColorMapActualMin()) * i +
       cm->getColorMapActualMin()) * scale);
     ss << str2 << " " << st->getValue(Units).toString();
-    text_.reset(ss.str(), textSize, Vector((displaySide == 0) ? 40. : 1., (displaySide == 0) ? 0. : 20., i));
+    text_.reset(ss.str(), textSize, Vector((displaySide == 0) ?
```
Comment (aylakhan) [inline src/Modules/Visualization/ShowColorMapModule.cc]: Magic numbers....

File: src/Interface/Modules/Render/ES/shaders/ColorMapLegend.vs [hunk 2]
```diff
@@ -42,18 +43,17 @@ varying float   vFieldData;
...
   bool half1 = uDisplayLength == 0.;
   vec3 newPos = ds?vec3(aPos.x, aPos.y, aPos.z):vec3(aPos.y,aPos.x,aPos.z);
   float x_scale = ds?(30.0 / uWindowWidth):(full?1.8:0.9);
   float y_scale = ds?(full?1.8:0.9):(30. * uAspectRatio / uWindowWidth);
-  float x_trans = ds?(-1.+(ex?.05:0.)):(full?-0.9:
-                  (half1?(ex?-.95:-1.):(ex?0.05:0.1)));
-  float y_trans = (!ds)?(-1.+(ex?.05:0.)):
-                  (full?-0.9:(half1?(ex?(-.95):
-                  -1.):(ex?0.05:0.1)));
+  float x_trans = (ds?(-1.):(full?-0.9:
```
Comment (aylakhan) [inline src/Interface/Modules/Render/ES/shaders/ColorMapLegend.vs]: Magic numbers...

File: src/Interface/Modules/Render/ES/shaders/ColorMapLegendText.vs [hunk 3]
```diff
@@ -42,17 +43,16 @@ varying vec4 fColor;
...
   bool full = uDisplayLength == 1.;
   bool half1 = uDisplayLength == 0.;
   vec3 newPos = aPos;
   float x_scale = 1.5 / uWindowWidth;
   float y_scale = 2.9 / (uWindowWidth / uAspectRatio);
-  float x_trans = ds?(-1.+(ex?.05:0.)):(-1.+(full?(aPos.z*1.8+.1):(aPos.z*.9+(half1?0.+(ex?.05:0.):
-                                       1.1+(ex?-.05:0.)))) - 15. / uWindowWidth);
-  float y_trans = ds?(-1.+(full?(aPos.z*1.8+.1):(aPos.z*.9+(half1?0.+(ex?.05:0.):
-                                       1.1+(ex?-.05:0.)))) - 2. / uWindowWidth):(-1.+(ex?.05:0.));
+  float x_trans = (ds?(-1.):(-1.+(full?(aPos.z*1.8+.1):(aPos.z*.9+(half1?0.:
```
Comment (aylakhan) [inline src/Interface/Modules/Render/ES/shaders/ColorMapLegendText.vs]: Magic numbers...

**Comment by dcwhite** [comment]
@brigb123 After you add the disabling commits to #903, I'll merge that one, then this one will need a master merge.

**Comment by dcwhite** [comment]
This appears to be working fine, but i do get these messages, not sure of priority:
Unable to find uniform with name: uCMInvert in shader with ID: 7
Unable to find uniform with name: uCMShift in shader with ID: 7
Unable to find uniform with name: uCMResolution in shader with ID: 7

**Comment by brigb123** [comment]
i must have missed deleting those uniform calls in the c++ code somewhere. i'll check that quickly

**Pull request merged**